pkg/datastore/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,336 @@
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+)
+
+// DynamoDB is wrapper for AWS dynamoDB client
+type DynamoDB struct {
+	client          *dynamodb.DynamoDB
+	profile         string
+	credentialsFile string
+	roleARN         string
+	tokenFile       string
+	endpoint        string
+
+	logger *zap.Logger
+}
+
+// Option for create DynamoDB typed instance
+type Option func(*DynamoDB)
+
+// WithLogger returns logger setup function
+func WithLogger(logger *zap.Logger) Option {
+	return func(s *DynamoDB) {
+		s.logger = logger
+	}
+}
+
+// WithCredentialsFile returns credentials infor setup function
+func WithCredentialsFile(profile, path string) Option {
+	return func(s *DynamoDB) {
+		s.profile = profile
+		s.credentialsFile = path
+	}
+}
+
+// WithTokenFile returns authenticate with token setup function
+func WithTokenFile(roleARN, path string) Option {
+	return func(s *DynamoDB) {
+		s.roleARN = roleARN
+		s.tokenFile = path
+	}
+}
+
+// NewDynamoDB returns new DynamoDB instance
+func NewDynamoDB(region, endpoint string, opts ...Option) (*DynamoDB, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region is required field")
+	}
+	if endpoint == "" {
+		return nil, fmt.Errorf("endpoint is required field")
+	}
+
+	s := &DynamoDB{
+		logger: zap.NewNop(),
+	}
+	for _, opt := range opts {
+		opt(s)
+	}
+
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a session: %w", err)
+	}
+	creds := credentials.NewChainCredentials(
+		[]credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{
+				Filename: s.credentialsFile,
+				Profile:  s.profile,
+			},
+			// roleSessionName specifies the IAM role session name to use when assuming a role.
+			// it will be generated automatically in case of empty string passed.
+			// ref: https://github.com/aws/aws-sdk-go/blob/0dd12669013412980b665d4f6e2947d57b1cd062/aws/credentials/stscreds/web_identity_provider.go#L116-L121
+			stscreds.NewWebIdentityRoleProvider(sts.New(sess), s.roleARN, "", s.tokenFile),
+			&ec2rolecreds.EC2RoleProvider{
+				Client: ec2metadata.New(sess),
+			},
+		},
+	)
+	cfg := aws.NewConfig().WithRegion(region).WithEndpoint(endpoint).WithCredentials(creds)
+	s.client = dynamodb.New(sess, cfg)
+
+	return s, nil
+}
+
+// Find implementation for DynamoDB
+func (s *DynamoDB) Find(ctx context.Context, kind string, opts datastore.ListOptions) (datastore.Iterator, error) {
+	expr, err := buildDynamoDBExpression(opts)
+	if err != nil {
+		s.logger.Error("failed to build query",
+			zap.String("kind", kind),
+			zap.Error(err),
+		)
+	}
+	input := &dynamodb.ScanInput{
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+		FilterExpression:          expr.Filter(),
+		TableName:                 aws.String(kind),
+	}
+	// TODO: Support pagination with ListOptions.Page and ListOptions.PageSize for DynamoDB.
+	// TODO: Support ordering with ListOptions.Orders for DynamoDB.
```

Comment:
It is fine to be a TODO. But can you tell me how to apply ordering with the Scan function?

Reply:
I would like to discuss that on the issue or PR for that so let's mark this conversation as resolved for now 😅